io.openems.edge.evcs.webasto.unite/src/io/openems/edge/evcs/webasto/unite/WebastoImpl.java

```diff
@@ -0,0 +1,289 @@
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.openems.common.exceptions.OpenemsError;
+import io.openems.common.exceptions.OpenemsException;
+import io.openems.edge.bridge.modbus.api.AbstractOpenemsModbusComponent;
+import io.openems.edge.bridge.modbus.api.BridgeModbus;
+import io.openems.edge.bridge.modbus.api.ModbusComponent;
+import io.openems.edge.bridge.modbus.api.ModbusProtocol;
+import io.openems.edge.bridge.modbus.api.element.SignedWordElement;
+import io.openems.edge.bridge.modbus.api.element.StringWordElement;
+import io.openems.edge.bridge.modbus.api.element.UnsignedDoublewordElement;
+import io.openems.edge.bridge.modbus.api.element.UnsignedWordElement;
+import io.openems.edge.bridge.modbus.api.task.FC3ReadRegistersTask;
+import io.openems.edge.bridge.modbus.api.task.FC4ReadInputRegistersTask;
+import io.openems.edge.bridge.modbus.api.task.FC6WriteRegisterTask;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.common.event.EdgeEventConstants;
+import io.openems.edge.common.taskmanager.Priority;
+import io.openems.edge.evcs.api.ChargeStateHandler;
+import io.openems.edge.evcs.api.Evcs;
+import io.openems.edge.evcs.api.EvcsPower;
+import io.openems.edge.evcs.api.ManagedEvcs;
+import io.openems.edge.evcs.api.Phases;
+import io.openems.edge.evcs.api.WriteHandler;
+import io.openems.edge.evcs.webasto.unite.api.Webasto;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Evcs.Webasto.Unite", //
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE //
+)
+@EventTopics({ //
+		EdgeEventConstants.TOPIC_CYCLE_BEFORE_PROCESS_IMAGE //
+})
+public class WebastoImpl extends AbstractOpenemsModbusComponent
+		implements OpenemsComponent, Webasto, Evcs, ManagedEvcs, EventHandler {
+
+	private final Logger log = LoggerFactory.getLogger(WebastoImpl.class);
+
+	@Reference
+	protected ConfigurationAdmin cm;
+
+	private Config config;
+
+	private int minCurrent;
+	private int maxCurrent;
+	private WebastoReadHandler readHandler;
+
+	@Reference
+	private EvcsPower evcsPower;
+
+	/**
+	 * Handles charge states.
+	 */
+	private final ChargeStateHandler chargeStateHandler = new ChargeStateHandler(this);
+
+	/**
+	 * Processes the controller's writes to this evcs component.
+	 */
+	private final WriteHandler writeHandler = new WriteHandler(this);
+
+	public WebastoImpl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				Webasto.ChannelId.values(), //
+				Evcs.ChannelId.values(), //
+				ModbusComponent.ChannelId.values(), //
+				ManagedEvcs.ChannelId.values() //
+		);
+	}
+
+	@Activate
+	void activate(ComponentContext context, Config config) throws ConfigurationException, OpenemsException {
+		super.activate(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm, "Modbus",
+				config.modbus_id());
+		this.config = config;
+
```

Comment:
Set Channels: 
this._setChargingType(ChargingType.AC);
this._setPowerPrecision(230); // 230 would be 1A steps

Reply:
This could be used in the future for having the real possible (steps) values in the UI (ion-range is limiting the values by the power precision)